File: mpi_c/pi/numapp.c
#include<math.h>
void numapp(double lo, double hi, int n, double *pres){
  double res, app, xi, xim, del, flo, fhi, fi, fim;
  int i, nl;
  flo =  sqrt(1.-lo*lo); fhi =  sqrt(1.-hi*hi);
  nl = n*(flo-fhi)+1;
  del = (hi-lo)/((double) nl);
  app = 0.; 
  for (i=0;i<nl;i++){
    xi = lo + (i+0.5)*del;
    app = app + sqrt(1.-xi*xi);
  }
  *pres = 4.*del*app;
  return;
}

File: pi/numapp.py
#!/usr/bin/python
import math 
import numpy as np

def numapp(lo,hi,n):
      flo = math.sqrt(1.-lo*lo)
      fhi = math.sqrt(1.-hi*hi)
      nl = int(n * (flo - fhi))
      delt = (hi-lo)/nl
      app = 0.0

      for i in range(0,nl):
         xi = lo + (i+0.5)*delt;
         app = app +  math.sqrt(1.-xi*xi)
      res = 4.*delt*app;
      return (res)
